sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/ExecutingSqlQueriesBase.cs

```diff
@@ -0,0 +1,87 @@
+        protected const string DiagnosticId = "S2077";
+        protected const string MessageFormat = "Make sure that executing SQL queries is safe here.";
+
+        protected InvocationTracker<TSyntaxKind> InvocationTracker { get; set; }
+
+        protected PropertyAccessTracker<TSyntaxKind> PropertyAccessTracker { get; set; }
+
+        protected ObjectCreationTracker<TSyntaxKind> ObjectCreationTracker { get; set; }
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            InvocationTracker.Track(context,
+                InvocationTracker.MatchSimpleNames(
+                    new MethodSignature(KnownType.Microsoft_EntityFrameworkCore_RelationalQueryableExtensions, "FromSql")),
+                ExceptWhen(OnlyParameterIsConstantOrInterpolatedString()),
+                ExceptWhen(InvocationTracker.FirstParameterIsConstant()));
+
+            InvocationTracker.Track(context,
+                InvocationTracker.MatchSimpleNames(
+                    new MethodSignature(KnownType.Microsoft_EntityFrameworkCore_RelationalDatabaseFacadeExtensions, "ExecuteSqlCommandAsync"),
+                    new MethodSignature(KnownType.Microsoft_EntityFrameworkCore_RelationalDatabaseFacadeExtensions, "ExecuteSqlCommand")),
+                ExceptWhen(OnlyParameterIsConstantOrInterpolatedString()));
+
+            PropertyAccessTracker.Track(context,
+                PropertyAccessTracker.MatchSimpleNames(
+                    new MethodSignature(KnownType.System_Data_Odbc_OdbcCommand, "CommandText"),
+                    new MethodSignature(KnownType.System_Data_OracleClient_OracleCommand, "CommandText"),
+                    new MethodSignature(KnownType.System_Data_SqlClient_SqlCommand, "CommandText"),
+                    new MethodSignature(KnownType.System_Data_SqlServerCe_SqlCeCommand, "CommandText")),
+                PropertyAccessTracker.MatchSet(),
+                ExceptWhen(PropertyAccessTracker.AssignedValueIsConstant()));
+
+            ObjectCreationTracker.Track(context,
+                ObjectCreationTracker.MatchConstructors(
+                    KnownType.Microsoft_EntityFrameworkCore_RawSqlString,
+                    KnownType.System_Data_SqlClient_SqlCommand,
+                    KnownType.System_Data_SqlClient_SqlDataAdapter,
+                    KnownType.System_Data_Odbc_OdbcCommand,
+                    KnownType.System_Data_Odbc_OdbcDataAdapter,
+                    KnownType.System_Data_SqlServerCe_SqlCeCommand,
+                    KnownType.System_Data_SqlServerCe_SqlCeDataAdapter,
+                    KnownType.System_Data_OracleClient_OracleCommand,
+                    KnownType.System_Data_OracleClient_OracleDataAdapter),
+                ObjectCreationTracker.FirstArgumentIs(KnownType.System_String),
+                ExceptWhen(ObjectCreationTracker.FirstArgumentIsConstant()));
+        }
+
+        private PropertyAccessCondition ExceptWhen(PropertyAccessCondition condition) =>
```

Comment:
I would like to use these elsewhere, not sure however where to put them... Perhaps we should create a base class...

Reply:
We could just create a static _Conditions_ class for now.